### PR TITLE
Folder restructure for evals/datasets/scoring

### PR DESCRIPTION
**Pull request by yanxi0830**
# What does this PR do?

- Restructure & rename evals/datasets/scoring related files. 
- Add braintrust fixture

#### existing test
```
pytest -v -s -m meta_reference_eval_together_inference_huggingface_datasetio eval/test_eval.py

pytest -v -s -m "localfs" datasetio/test_datasetio.py

pytest -v -s -m meta_reference_scoring_fireworks_inference scoring/test_scoring.py
```

#### new test
```
pytest -v -s -m braintrust_scoring_together_inference scoring/test_scoring.py
```

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Ran pre-commit to handle lint / formatting issues.
- [ ] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [ ] Updated relevant documentation.
- [x] Wrote necessary unit or integration tests.

### CODE REVIEW

File: llama_stack/providers/inline/datasetio/localfs/config.py [hunk 1]
```diff
@@ -6,4 +6,4 @@
 from llama_stack.apis.datasetio import *  # noqa: F401, F403
 
 
-class MetaReferenceDatasetIOConfig(BaseModel): ...
+class LocalfsDatasetIOConfig(BaseModel): ...
```
Comment (raghotham) [inline llama_stack/providers/inline/datasetio/localfs/config.py:9]: LocalFS?

Comment (yanxi0830) [reply llama_stack/providers/inline/datasetio/localfs/config.py]: done!

File: llama_stack/providers/inline/scoring/braintrust/braintrust.py [hunk 2]
```diff
@@ -16,9 +16,8 @@
 from autoevals.llm import Factuality
 from autoevals.ragas import AnswerCorrectness
 from llama_stack.providers.datatypes import ScoringFunctionsProtocolPrivate
-from llama_stack.providers.inline.meta_reference.scoring.scoring_fn.common import (
-    aggregate_average,
-)
+
+from ..meta_reference.scoring.scoring_fn.common import aggregate_average
```
Comment (raghotham) [inline llama_stack/providers/inline/scoring/braintrust/braintrust.py:20]: this seems awkward. can we move common stuff to providers/utils or something?

File: llama_stack/providers/inline/scoring/meta_reference/scoring_fn/llm_as_judge_scoring_fn.py [hunk 3]
```diff
@@ -4,20 +4,15 @@
...
 import re
 
-from llama_stack.providers.inline.meta_reference.scoring.scoring_fn.common import (
-    aggregate_average,
-)
-from llama_stack.providers.inline.meta_reference.scoring.scoring_fn.fn_defs.llm_as_judge_8b_correctness import (
-    llm_as_judge_8b_correctness,
-)
+from .common import aggregate_average
+from .fn_defs.llm_as_judge_8b_correctness import llm_as_judge_8b_correctness
```
Comment (raghotham) [inline llama_stack/providers/inline/scoring/meta_reference/scoring_fn/llm_as_judge_scoring_fn.py:15]: why is model name in the module name? I thought we were making the model itself a parameter for the judge?

File: llama_stack/providers/registry/datasetio.py [hunk 4]
```diff
@@ -13,10 +13,10 @@ def available_providers() -> List[ProviderSpec]:
     return [
         InlineProviderSpec(
             api=Api.datasetio,
-            provider_type="meta-reference",
+            provider_type="localfs",
             pip_packages=["pandas"],
-            module="llama_stack.providers.inline.meta_reference.datasetio",
-            config_class="llama_stack.providers.inline.meta_reference.datasetio.MetaReferenceDatasetIOConfig",
+            module="llama_stack.providers.inline.datasetio.localfs",
```
Comment (raghotham) [inline llama_stack/providers/registry/datasetio.py:18]: do we want to have an inline/datasetio/pandas as a provider as well? or are we saying that pandas is the interface between datasetio and rest of the stack?

Comment (yanxi0830) [reply llama_stack/providers/registry/datasetio.py:18]: pandas is a library that is used for both huggingface provider and localfs provider.

File: llama_stack/providers/inline/scoring/meta_reference/scoring_fn/llm_as_judge_scoring_fn.py (hunk 3 above)
Comment (yanxi0830) [reply llama_stack/providers/inline/scoring/meta_reference/scoring_fn/llm_as_judge_scoring_fn.py:15]: oh my bad, I need to delete this

File: llama_stack/providers/inline/scoring/meta_reference/scoring_fn/fn_defs/llm_as_judge_base.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from llama_stack.apis.scoring_functions import *  # noqa: F401, F403
```
Comment (raghotham) [inline llama_stack/providers/inline/scoring/meta_reference/scoring_fn/fn_defs/llm_as_judge_base.py:7]: can we not do import *? there were complaints of lack of readability.

Comment (yanxi0830) [reply llama_stack/providers/inline/scoring/meta_reference/scoring_fn/fn_defs/llm_as_judge_base.py:7]: Updated this, we can codemod this later in a separate PR.

**Pull request merged**